src/context.ts

```diff
@@ -75,3 +77,19 @@ export class OperonContextImpl implements OperonContext {
     return value as T;
   }
 }
+
+export interface InitContext extends OperonContext {
+  readonly userDatabase: UserDatabase;
```

Comment:
We shouldn't expose this directly.  Need a better abstraction for schema management.

Reply:
I also don't think this should extend `OperonContext`, as a lot of the fields there don't make sense here.  The only methods from `OperonContext` this needs are `getConfig` and logger access.

Reply:
As you mentioned in the meeting, one idea is to expose a client similar to the one we had in TransactionContext (but the init code may not be transactional because create/drop tables are not).

Reply:
I don't see why you shouldn't have the user database.  Do you understand the requirements?  Would you tell me the same thing if I was an actual customer?

It should also have the logger.

Reply:
@chuck-dbos I mean we don't need to expose our own `UserDatabase` interface, but we should give users full power of the DB client they chose. `UserDatabase` interface is an implementation suited for our internal operations, which has some weird interface methods. We might want to expose something else that is powerful but more natural to users (like a pg-pool or some equivalent in ORMs). Do you have any suggestion?

Reply:
We need a base Context class that has only logger , appconfig and say the dbclient

Reply:
Init code needs database connection just like other code. For transactions we give the dbclient via the TransactionContext. But TransactionContext gets the dbclient in a round about way -- It is inserted in there when workflow.transaction calls user_database.transaction. For init code , what if we just took the dbclient from user_database and gave it to init code via some context ?

Reply:
I was aiming for consistency with what you can do now:
   return operon.userDatabase.createSchema();

But the other methods on userDatabase that are useful are:
    getName(): UserDatabaseName;
    query<R>(sql: string, ...params: any[]): Promise<R[]>;
    dropSchema(): Promise<void>;
    transaction (of some sort)

Imagine the scenario is that you have an empty "serverless" database provided to you and you need to create the schema and add the superuser / prepopulate a few other rows, like you would now.  And the API ought to match other code.

Reply:
Another inspiration comes from the test runtime API... I would think these the useful methods therein:

    getConfig<T>(key: string): T | undefined;
    getConfig<T>(key: string, defaultValue: T): T;
    queryUserDB<R>(sql: string, ...params: any[]): Promise<R[]>;
    createUserSchema(): Promise<void>;
    dropUserSchema(): Promise<void>;

But also logging of some sort, to report any trouble.